packages/state-transition/src/util/shuffle.ts

```diff
@@ -70,7 +70,7 @@ function setPositionUint32(value: number, buf: Buffer): void {
 }
 
 // Shuffles or unshuffles, depending on the `dir` (true for shuffling, false for unshuffling
-function innerShuffleList(input: ValidatorIndex[], seed: Bytes32, dir: boolean): void {
+function innerShuffleList(input: Uint32Array | ValidatorIndex[], seed: Bytes32, dir: boolean): void {
```

Comment:
Is the `ValidatorIndex[]` type correct here? Same in the other function definitions.

Reply:
Yeah, its intentional, its just a wider type than necessary, allows the tests to continue to pass unchanged. Really the implementation relies on an interface like:
```ts
interface Shuffleable {
  [index: number]: number
  length: number
}
```